src/biotite/database/uniprot/download.py

```diff
@@ -0,0 +1,190 @@
+        If no `target_path` was given, the file contents are stored in
+        either `StringIO` or `BytesIO` objects.
+
+    Warnings
+    --------
+    Even if you give valid input to this function, in rare cases the
+    database might return no or malformed data to you.
+    In these cases the request should be retried.
+    When the issue occurs repeatedly, the error is probably in your
+    input.
+
+    Examples
+    --------
+
+    >>> import os.path
+    >>> file = fetch("P12345", "fasta", path_to_directory)
+    >>> print(os.path.basename(file))
+    P12345.fasta
+    >>> files = fetch(["P12345", "Q8K9I1"], "fasta", path_to_directory)
+    >>> print([os.path.basename(file) for file in files])
+    ['P12345.fasta', 'Q8K9I1.fasta']
+    """
+    # If only a single UID is present,
+    # put it into a single element list
+    if isinstance(uids, str):
+        uids = [uids]
+        single_element = True
+    else:
+        single_element = False
+    # Create the target folder, if not existing
+    if target_path is not None and not isdir(target_path):
+        os.makedirs(target_path)
+    files = []
+    for i, id in enumerate(uids):
+        db_name = _get_database_name(id)
+        # Verbose output
+        if verbose:
+            print(f"Fetching file {i + 1:d} / {len(uids):d} ({id})...",
+                  end="\r")
+        # Fetch file from database
+        if target_path is not None:
+            file = join(target_path, id + "." + format)
+        else:
+            # 'file = None' -> store content in a file-like object
+            file = None
+        if file is None \
+                or not isfile(file) \
+                or getsize(file) == 0 \
+                or overwrite:
+            if format == "fasta":
+                r = requests.get(fetch_url + db_name + "/" + id + ".fasta")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "gff":
+                r = requests.get(fetch_url + db_name + "/" + id + ".gff")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "txt":
+                r = requests.get(fetch_url + db_name + "/" + id + ".txt")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "xml":
+                r = requests.get(fetch_url + db_name + "/" + id + ".xml")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "rdf":
+                r = requests.get(fetch_url + db_name + "/" + id + ".rdf")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
+            elif format == "tab":
+                r = requests.get(fetch_url + db_name + "/" + id + ".tab")
+                content = r.text
+                _assert_valid_file(r.status_code, id, format)
```

Comment:
This if-else-block contains quite a lot of repetitions. Alternatively, you could check for a valid file format at first and then handle the request for any format using the same code.